From 57abf55bba172dfb4301674a75c1f663a70741d0 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Thu, 25 May 2023 10:16:08 +0200
Subject: [PATCH 3/9] fix the CreateHardLinkW signature to match the real
 function

The `ret` variable was already using BOOL instead of BOOLEAN.
---
 libarchive/archive_write_disk_windows.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libarchive/archive_write_disk_windows.c b/libarchive/archive_write_disk_windows.c
index b42010d4..839bc295 100644
--- a/libarchive/archive_write_disk_windows.c
+++ b/libarchive/archive_write_disk_windows.c
@@ -580,7 +580,7 @@ la_GetFunctionKernel32(const char *name)
 static int
 la_CreateHardLinkW(wchar_t *linkname, wchar_t *target)
 {
-	static BOOLEAN (WINAPI *f)(LPWSTR, LPWSTR, LPSECURITY_ATTRIBUTES);
+	static BOOL (WINAPI *f)(LPCWSTR, LPCWSTR, LPSECURITY_ATTRIBUTES);
 	BOOL ret;
 
 #if _WIN32_WINNT < _WIN32_WINNT_XP
-- 
2.37.3.windows.1

